ONX.CRM.BLL/Services/StudentRequestService.cs

```diff
@@ -28,5 +32,24 @@ public void AssignRequestToGroups(IEnumerable<StudentRequest> requests, int grou
                 _repository.Delete(request.Id);
             }
         }
+
+        public async Task<IEnumerable<StudentRequest>> GetRequestsByCourseId(int courseId)
+        {
+            var requestsList = (await _repository.GetAllAsync())
+                .Where(r => r.CourseId == courseId);
+            return requestsList;
+        }
+
+
+        public async Task<Dictionary<int, string>> GetCoursesForDropdown()
```

Comment:
Dropdown is part of Presentation layer. As a result if such method, Business logis is logically depends on Presentation layer that is architectural issue